yash.c

```diff
@@ -256,20 +258,33 @@ struct input_file_info_T *new_input_file_info(int fd, size_t bufsize)
     return info;
 }
 
-/* Executes "$HOME/.yash_profile". */
+/* Executes "$XDG_CONFIG_HOME/yash/profile", "$HOME/.config/yash/profile" or
+ * "$HOME/.yash_profile" as a fallback.
+ */
 void execute_profile(const wchar_t *profile)
 {
     if (profile != NULL)
         execute_file(profile);
```

Comment:
When the `profile` is specified, we should only execute it, so:

```suggestion
    if (profile != NULL) {
        execute_file(profile);
        return;
    }
```